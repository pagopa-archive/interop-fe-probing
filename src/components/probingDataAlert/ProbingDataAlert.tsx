import { Alert, Grid } from '@mui/material'
import { useTranslation } from 'react-i18next'
import { useState, useEffect } from 'react'

interface ProbingData {
  probingEnabled: boolean
  state: string
  eserviceActive: boolean
}

export const ProbingDataAlert: React.FC<ProbingData> = ({
  probingEnabled,
  state,
  eserviceActive,
}) => {
  const [message, setMessage] = useState('')

  const { t } = useTranslation(['detailsPage'])

  useEffect(() => {
    if (probingEnabled === false && state === 'n/d') {
      setMessage('monitoringSystemSuspendedMessage')
    } else if (state === 'offline' && !eserviceActive) {
      setMessage('versionSuspendedMessage')
    } else if (state === 'offline' && eserviceActive) {
      setMessage('eserviceNotAnswerMessage')
    } else {
      setMessage('')
    }
  }, [probingEnabled, state, eserviceActive])

  return (
    <>
      {message !== '' ? (
        <Grid item container>
          <Alert
            sx={{
              width: '100%',
            }}
            severity="warning"
          >
            {t(message)}
          </Alert>
        </Grid>
      ) : null}
    </>
  )
}
