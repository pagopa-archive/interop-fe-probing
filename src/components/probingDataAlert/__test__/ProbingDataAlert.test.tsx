import { describe, expect, test } from 'vitest'
import { render } from '@testing-library/react'
import { ProbingDataAlert } from '../ProbingDataAlert'

const mainData = {
  eserviceName: 'Probing test 3',
  producerName: 'Comune di Milano',
  versionNumber: '7',
  pollingFrequency: 5,
}

describe('ProbingDataAlert', () => {
  test('render component case 1', () => {
    const { container } = render(
      <ProbingDataAlert probingEnabled={true} state={'OFFLINE'} eserviceActive={false} />
    )
    expect(container).toMatchSnapshot()
  })

  test('render component case 2', () => {
    const { container } = render(
      <ProbingDataAlert probingEnabled={true} state={'OFFLINE'} eserviceActive={true} />
    )
    expect(container).toMatchSnapshot()
  })

  test('render component case 3', () => {
    const { container } = render(
      <ProbingDataAlert probingEnabled={false} state={'N/D'} eserviceActive={false} />
    )
    expect(container).toMatchSnapshot()
  })
})
