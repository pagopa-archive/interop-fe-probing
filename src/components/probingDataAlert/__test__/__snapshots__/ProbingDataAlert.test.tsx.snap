// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProbingDataAlert > render component 1`] = `<div />`;

exports[`ProbingDataAlert > render component 2`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item css-1f064cs-MuiGrid-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardWarning MuiAlert-standard css-1j916qe-MuiPaper-root-MuiAlert-root"
      role="alert"
    >
      <div
        class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
          data-testid="ReportProblemOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
      >
        eserviceNotAnswerMessage
      </div>
    </div>
  </div>
</div>
`;

exports[`ProbingDataAlert > render component 3`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item css-1f064cs-MuiGrid-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardWarning MuiAlert-standard css-1j916qe-MuiPaper-root-MuiAlert-root"
      role="alert"
    >
      <div
        class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
          data-testid="ReportProblemOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
      >
        monitoringSystemSuspendedMessage
      </div>
    </div>
  </div>
</div>
`;

exports[`ProbingDataAlert > render component case 1 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item css-1f064cs-MuiGrid-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardWarning MuiAlert-standard css-1j916qe-MuiPaper-root-MuiAlert-root"
      role="alert"
    >
      <div
        class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
          data-testid="ReportProblemOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
      >
        versionSuspendedMessage
      </div>
    </div>
  </div>
</div>
`;

exports[`ProbingDataAlert > render component case 2 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item css-1f064cs-MuiGrid-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardWarning MuiAlert-standard css-1j916qe-MuiPaper-root-MuiAlert-root"
      role="alert"
    >
      <div
        class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
          data-testid="ReportProblemOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
      >
        eserviceNotAnswerMessage
      </div>
    </div>
  </div>
</div>
`;

exports[`ProbingDataAlert > render component case 3 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item css-1f064cs-MuiGrid-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardWarning MuiAlert-standard css-1j916qe-MuiPaper-root-MuiAlert-root"
      role="alert"
    >
      <div
        class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
          data-testid="ReportProblemOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
      >
        monitoringSystemSuspendedMessage
      </div>
    </div>
  </div>
</div>
`;
